Components/Sources/Components/Components/Suggested Edits/Image Recommendations/WKImageRecommendationBottomSheetView.swift

```diff
@@ -334,28 +298,58 @@ public class WKImageRecommendationBottomSheetView: WKComponentView {
         toolbar.barTintColor = theme.midBackground
+        textView.linkTextAttributes = [.foregroundColor: theme.link,
+                                       .font: WKFont.for(.boldCallout)
+        ]
     }
 
     private func configure() {
         imageView.image = viewModel.imageThumbnail
-        textView.text = viewModel.imageDescription
         titleLabel.text = viewModel.headerTitle
-        imageLinkButton.setAttributedTitle(getImageLinkButtonTitle(), for: .normal)
+        textView.delegate = self
+        textView.attributedText = getTextViewAttributedString()
+        updateColors()
     }
 
-    private func getImageLinkButtonTitle() -> NSMutableAttributedString {
+    private func getTextViewAttributedString() -> NSMutableAttributedString {
         let attributedString = NSMutableAttributedString()
-        if let imageAttachment = WKIcon.externalLink {
-            let attachment = NSTextAttachment(image: imageAttachment)
-            attributedString.append(NSAttributedString(string: viewModel.imageTitle))
-            attributedString.append(NSAttributedString(string: "  "))
-            attributedString.append(NSAttributedString(attachment: attachment))
+
+        let linkAttributedString = NSMutableAttributedString(string: viewModel.imageTitle)
+
+        let attachment = NSTextAttachment()
+        if let image = WKIcon.externalLink {
+            let tintedImage = image.withTintColor(theme.link)
+            attachment.image = tintedImage
+        }
+        let attachmentAttributedString = NSMutableAttributedString(string: " ")
+        attachmentAttributedString.append(NSAttributedString(attachment: attachment))
+        attachmentAttributedString.addAttributes([.foregroundColor: theme.link], range: NSRange(location: 0, length: attachmentAttributedString.length))
+
+        attributedString.append(linkAttributedString)
+        attributedString.append(attachmentAttributedString)
+        attributedString.append(NSAttributedString(string: "\n\n"))
+
+        if let description = viewModel.imageDescription {
+            let descriptionAttributes = [NSAttributedString.Key.font: WKFont.for(.callout),
+                                         NSAttributedString.Key.foregroundColor: theme.text]
+            let descriptionAttributedString = NSMutableAttributedString(string: description, attributes: descriptionAttributes)
+            attributedString.append(descriptionAttributedString)
+            attributedString.append(NSAttributedString(string: "\n\n"))
         }
+
+        let reasonAttributes = [NSAttributedString.Key.font: WKFont.for(.callout),
+                                NSAttributedString.Key.foregroundColor: theme.secondaryText]
+        let reasonAttributedString = NSMutableAttributedString(string: viewModel.reason, attributes: reasonAttributes)
+        attributedString.append(reasonAttributedString)
+        attributedString.append(NSAttributedString(string: "\n\n"))
```

Comment:
This is a hack not to cut the text. I've experimented with the hugging and compression text view props, masksToBounds, clipsToBounds, and sizeToFit. I have the attributed string before setting the textView layout; however, none of that worked to calculate and set the correct size. Anything I might have overlooked?

Reply:
@mazevedofs I played around with this to try and fix it. I think the key was

1. Constraining the text view to the [bottom](https://github.com/wikimedia/wikipedia-ios/pull/4793/commits/96384feb9bd80d04f353da30db9e9065bc66ddce#diff-b62ea79cb164ad06e0d4082804e46cba05300ed37bcbddbfc536352fda8a2254R264) of the of the container, so that the container expands to the full height of the text view content
2. Once I did that, the header stack view height grew super tall. Setting the alignment of it [here](https://github.com/wikimedia/wikipedia-ios/pull/4793/commits/96384feb9bd80d04f353da30db9e9065bc66ddce#diff-b62ea79cb164ad06e0d4082804e46cba05300ed37bcbddbfc536352fda8a2254R37) to `.fill` shrunk it back down to just the contents 🤷‍♀️
3. I also added some content hugging / compression priorities as I was trying different things. I'm not sure if those changes are necessary.